policy-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/policy/kafka/KafkaContractTest.java

```diff
@@ -289,6 +298,22 @@ void testNoSuchPolicyExceptionIsThrown() {
         results.removeFirst();
         results.removeLast();
         assertThat(results).isEmpty();
+
+        assertAll("Asserting on the error topic",
+                // One error is produced
+                () -> assertThat(errorResults)
+                        .hasSize(1),
+
+                // The error has the relevant headers, including the token
+                () -> assertThat(errorResults)
+                        .allSatisfy(result ->
+                                assertThat(result.headers().lastHeader(Token.HEADER).value())
+                                        .isEqualTo(ContractTestData.REQUEST_TOKEN.getBytes())),
+
+                // The error has a message that contains the throwable exception, and the message
+                () -> assertThat(errorResults.get(0).value().get("error").get("message").asText())
+                        .isEqualTo(NoSuchPolicyException.class.getName() + ": No Resource Rules found for the resource")
+        );
```

Comment:
```suggestion
        assertThat(errorResults)
                .hasSize(1)
                .allSatisfy(result -> assertThat(result.headers().lastHeader(Token.HEADER).value())
                        .isEqualTo(ContractTestData.REQUEST_TOKEN.getBytes()))
                .allSatisfy(result -> assertThat(errorResults.get(0).value().get("error").get("message").asText())
                        .isEqualTo(NoSuchPolicyException.class.getName() + ": No Resource Rules found for the resource"));
```

Reply:
I want to keep the comments.  I think it helps in reading the code.